.github/workflows/publish.yml

```diff
@@ -0,0 +1,66 @@
+# This workflow will build a package using Maven and then publish it to GitHub packages when a release is created
+# For more information see: https://github.com/actions/setup-java#apache-maven-with-a-settings-path
+
+name: Publish to Central Maven Repo
+
+on:
+  release:
+    types: [created]
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout the code
+      uses: actions/checkout@v2
+
+    - name: Configure java
+      uses: actions/setup-java@v1
+      with:
+        java-version: 11
+        server-id: github # Va  lue of the distributionManagement/repository/id field of the pom.xml
+        settings-path: ${{ github.workspace }} # location for the settings.xml file
+
+    - name: Build and run tests
+      run: mvn -B package 
+
+    - name: Run the integration tests
+      run: mvn -B package 
+
+    - name: Retrieve Credentials for Publishing Maven package
+      uses: say8425/aws-secrets-manager-actions@v1
```

Comment:
one unfortunate thing about this action is that it does not support assuming a role.

I wasted a LOT of time trying to use https://github.com/aws-actions/configure-aws-credentials and then a fork of this action to use an assumed role for credentials, but just could not get it to work.

so for now I attached a policy giving access to the secret directly to the user on whose behalf this is running.

I could spend a bit more time trying to get my fork to work by calling the assume role API directly in the action instead of relying on https://github.com/aws-actions/configure-aws-credentials, OTOH I am not sure its worth it, and the very least i didnt want to be blocked in the short term